src/parser.rs

```diff
@@ -532,6 +540,21 @@ impl<'a> Parser<'a> {
         Ok(expr_list)
     }
 
+    fn parse_limit(&mut self) -> Result<Option<Box<ASTNode>>, ParserError> {
+        if self.index < self.tokens.len() {
```

Comment:
You might want to try using `peek_token()` or `next_token()` here or even `if parse_keyword("ALL")`

Reply:
This is looking great though  ... just some minor things that could be improved 👍